workitem/workitem.go

```diff
@@ -53,3 +57,112 @@ func (wi WorkItem) GetLastModified() time.Time {
 	log.Debug(nil, map[string]interface{}{"wi_id": wi.ID}, "Last modified value: %v", lastModified)
 	return *lastModified
 }
+
+// ChangeSet derives a changeset between this workitem and a given workitem.
+func (wi WorkItem) ChangeSet(older convert.ChangeDetector) ([]convert.Change, error) {
+	if older == nil {
+		// this is changeset for a new ChangeDetector, report all observed attributes to
+		// the change set. This needs extension once we support more attributes.
+		changeSet := []convert.Change{
+			{
+				AttributeName: SystemState,
```

Comment:
Why construct the change set by hand and not by iterating over all the fields of the WIT and using the `wi.Fields[fieldName]` ad the `NewValue`?

Reply:
This is an MVP. If I would think a bit ahead here, I would always think that the change should not only touch the Fields but also other data on the entity. As because we don't know that yet, this is an MVP that has the big words "put extensions here" on it.

Reply:
Fair enough.